### PR TITLE
Suppress 3rdParty warnings on Clang

### PR DESCRIPTION
**Pull request by goedderz**
### Scope & Purpose

Suppress some Clang warnings in 3rdParty libraries to reduce noise while building.

- [ ] :hankey: Bugfix (requires CHANGELOG entry)
- [ ] :pizza: New feature (requires CHANGELOG entry, feature documentation and release notes)
- [ ] :fire: Performance improvement
- [X] :hammer: Refactoring/simplification
- [ ] :book: CHANGELOG entry made

#### Backports:

- [X] No backports required
- [ ] Backports required for: *(Please specify versions and link PRs)*

### Testing & Verification

- [X] This change is a trivial rework / code cleanup without any test coverage.
- [ ] The behavior in this PR was *manually tested*
- [ ] This change is already covered by existing tests, such as *(please describe tests)*.
- [ ] This PR adds tests that were used to verify all changes:
  - [ ] Added new C++ **Unit tests**
  - [ ] Added new **integration tests** (e.g. in shell_server / shell_server_aql)
  - [ ] Added new **resilience tests** (only if the feature is impacted by failovers)
- [ ] There are tests in an external testing repository:
- [ ] I ensured this code runs with ASan / TSan or other static verification tools

### CODE REVIEW

File: 3rdParty/V8/CMakeLists.txt [hunk 1]
```diff
@@ -353,6 +353,9 @@ else ()
     set(V8_CFLAGS   "${V8_CFLAGS} -fno-delete-null-pointer-checks")
     set(V8_CXXFLAGS "${V8_CXXFLAGS} -fno-delete-null-pointer-checks")
   endif()
+  if (CMAKE_CXX_COMPILER_ID STREQUAL "Clang")
+      set(V8_CXXFLAGS "${V8_CXXFLAGS} -Wno-suggest-override -Wno-implicit-const-int-float-conversion -Wno-non-virtual-dtor -Wno-final-dtor-non-final-class -Wno-implicit-const-int-float-conversion")
+  endif()
```
Comment (dothebart) [inline 3rdParty/V8/CMakeLists.txt:359]: pls also set V8_CFLAGS - just to have this symetric.

File: 3rdParty/s2geometry/CMakeLists.txt [hunk 2]
```diff
@@ -3,6 +3,10 @@
 set (PYTHON_INCLUDE_DIR "")
 project(s2-geometry C CXX)
 
+if (CMAKE_CXX_COMPILER_ID STREQUAL "Clang")
```
Comment (dothebart) [inline 3rdParty/s2geometry/CMakeLists.txt:6]: will this also work for #include statements?

**Review by dothebart — APPROVED** [review]
please also set the cflags, otherwise LGTM

File: 3rdParty/V8/CMakeLists.txt (hunk 1 above)
Comment (goedderz) [reply 3rdParty/V8/CMakeLists.txt:360]: Done.

File: 3rdParty/s2geometry/CMakeLists.txt (hunk 2 above)
Comment (goedderz) [reply 3rdParty/s2geometry/CMakeLists.txt:6]: Sadly, no. This can only suppress warnings in the s2 cpp files, not when we include the headers somewhere.

**Review by dothebart — APPROVED** [review]

**Pull request merged**